pkg/doc/verifiable/example_presentation_test.go

```diff
@@ -563,3 +567,187 @@ func ExamplePresentation_MarshalledCredentials() {
 	//	]
 	//}
 }
+
+//nolint:lll,gocyclo
+func ExamplePresentation_AddLinkedDataProof() {
+	// 1. ISSUER issues a VC.
+	vcToIssue := `
+{
+  "@context": [
+    "https://www.w3.org/2018/credentials/v1",
+    "https://www.w3.org/2018/credentials/examples/v1"
+  ],
+  "credentialSubject": {
+    "degree": {
+      "type": "BachelorDegree",
+      "university": "MIT"
+    },
+    "id": "did:example:ebfeb1f712ebc6f1c276e12ec21",
+    "name": "Jayden Doe",
+    "spouse": "did:example:c276e12ec21ebfeb1f712ebc6f1"
+  },
+  "expirationDate": "2020-01-01T19:23:24Z",
+  "id": "http://example.edu/credentials/1872",
+  "issuanceDate": "2010-01-01T19:23:24Z",
+  "issuer": {
+    "id": "did:example:76e12ec712ebc6f1c221ebfeb1f",
+    "name": "Example University"
+  },
+  "type": [
+    "VerifiableCredential",
+    "UniversityDegreeCredential"
+  ]
+}
+`
+
+	issuedVC, err := verifiable.ParseUnverifiedCredential([]byte(vcToIssue),
+		verifiable.WithJSONLDDocumentLoader(getJSONLDDocumentLoader()))
+	if err != nil {
+		panic(fmt.Errorf("failed to decode VC JSON: %w", err))
+	}
+
+	issuerSigner := signature.GetEd25519Signer(issuerPrivKey, issuerPubKey)
+
+	err = issuedVC.AddLinkedDataProof(&verifiable.LinkedDataProofContext{
+		Created:                 &issued,
+		SignatureType:           "Ed25519Signature2018",
+		Suite:                   ed25519signature2018.New(suite.WithSigner(issuerSigner)),
+		SignatureRepresentation: verifiable.SignatureJWS,
+		VerificationMethod:      "did:example:123456#key1",
+	}, jsonld.WithDocumentLoader(getJSONLDDocumentLoader()))
+	if err != nil {
+		panic(fmt.Errorf("failed to add linked data proof: %w", err))
+	}
+
+	// 2. ISSUER creates a VP with the VC enclosed.
+	vpStr := `
```

Comment:
@kdimak in this case, since we already have the VC, isn't the proper way to begin creating a VP with the VC by first calling `vc.Presentation()` ?

Reply:
As a user, I'm trying to understand how to best use this library

Reply:
Yes, that is correct, we could use `vc.Presentation()`. I think we should use this when we have a parsed VC (using `ParseCredential()` or `ParseUnverifiedCredential()` or constructed manually). From the other hand, if we have VC in JSON form (or in JWT form), then we can construct the VP and then pass one or more credentials in text form using `vp.SetCredentials()`.

Reply:
And I will change the example to use `vc.Presentation()`, this is more relevant use case. Thanks @llorllale !